### ISSUE TITLE
Remove old cred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11066]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1440|
|Type|enhancement|
|Created|2006-01-23 04:42:41Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1440 1440
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1137991361000000 1137991361000000
changetime__1149477137000000 1149477137000000
version__ 
owner__ 
cc__exarkun cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `assigned`|
|-|-|

This is mostly done.  The only thing left to do is re-implement four PB tests which used deprecated APIs but which tested non-deprecated functionality.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Ready for review in remove-old-cred-1440

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

Mr. Wiki say: [branches/remove-old-cred-1440](branches/remove-old-cred-1440)  (make a link for the lazy!)

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to @exarkun|
|-|-|

In `twisted/test/test_pb.py`:

```
-    def testImmediateClose(self):
-        cc = protocol.ClientCreator(reactor, protocol.Protocol)
-        d = cc.connectTCP("127.0.0.1", self.portno)
-        d.addCallback(lambda p: p.transport.loseConnection())
-        d = defer.Deferred()
-        # clunky, but it works
-        reactor.callLater(0.1, d.callback, None)
-        return d
```

I don't see a replacement for this test.  It's obviously a crappy test, but still... it would be nice to have a test that an immediate disconnect doesn't cause an unhandled exception.  It can probably be a proper unit test that just instantiates the relevant protocol and calls `connectionMade` then `connectionLost`, never touching the reactor, rather than the vile thing it is now.

(I don't see a replacement for many tests in that file, but unlike the others, this one doesn't have any supressions for deprecation warnings)

`testClientConnectionLost` is horribly complicated for what sounds from the docstring like a fairly simple test, but that seems hard to avoid (and isn't a new problem anyway).  Surprisingly though it contains no `self.assert`/`failUnless...` calls.  It'd be worth putting an `self.failUnless(isinstance(anotherRootObj, pb.RemoteReference)` in `gotAnotherRootObj` just for the sake of clarity.

Anyway, I'm **very** happy with this.  It deletes over 3000 lines of cruft!  You've just wiped thousands of dollars of value off our `sloccount` results...

The branch is good enough to merge as-is, but consider my suggestions.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

I added a test for immediate close that does what you suggested.  As I was going to merge this, I realized I had missed some things.  Two files in conch, authorizer.py and identity.py, as well as a pile of pb perspective documentation.  I deleted all of these and removed the references to the documentation from the various index files.  I also deleted a couple exception classes from twisted/cred/error.py

I don't _think_ there is any more old cred stuff, but it's hard to tell.

Another pass of review would be appreciated.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to spiv<br>spiv set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to @exarkun<br>spiv set status to `new`|
|-|-|

You broke the docs build: http://twistedmatrix.com/buildbot/full-2.3/builds/1591/step-process-docs/0

Otherwise, this looks fine to me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17081]) Merge remove-old-cred-1440

Author: exarkun, radix
Reviewer: spiv
Fixes [#11066](https://github.com/twisted/twisted/issues/11066)

This removes the old authentication and authorization APIs from twisted.cred
as well as the old Application and Service APIs from twisted.internet.  Compatibility
support code for these APIs has also been removed, as have their unit tests and
documentation which referred to them.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
